pkg/deployable/proxy.go

```diff
@@ -246,8 +241,13 @@ func (ph *proxyHandler) getProxyHandler(token string, clusterConfig *rest.Config
 				reverseProxy.ServeHTTP(rw, r)
 				return
 			}
-
-			b, err = ph.translateBody(b)
+			var translator translator
+			if r.Header.Get("Content-Type") == "application/json" {
+				translator = NewJSONTranslator(b, ph.Name, ph.DivertDriver)
+			} else {
+				translator = NewProtobufTranslator(b, ph.Name, ph.DivertDriver)
+			}
```

Comment:
Instead of creating the translators inline, we can have a map [string]translator, where the key is the content-type, so we only created them once, but used as many times as needed